modules/product/src/main/java/com/opengamma/strata/product/swap/type/ImmutableFixedInflationSwapConvention.java

```diff
@@ -0,0 +1,639 @@
+ * <p>
+ * The convention is defined by four key dates.
+ * <ul>
+ * <li>Trade date, the date that the trade is agreed
+ * <li>Spot date, the base for date calculations, typically 2 business days after the trade date
+ * <li>Start date, the date on which the interest calculation starts, often the same as the spot date
+ * <li>End date, the date on which the interest calculation ends, typically a number of years after the start date
+ * </ul>
+ */
+@BeanDefinition
+public final class ImmutableFixedInflationSwapConvention
+    implements FixedInflationSwapConvention, ImmutableBean, Serializable {
+
+  /**
+   * The convention name, such as 'USD-FIXED-6M-LIBOR-3M'.
+   */
+  @PropertyDefinition(validate = "notNull", overrideGet = true)
+  private final String name;
+  /**
+   * The market convention of the fixed leg.
+   */
+  @PropertyDefinition(validate = "notNull", overrideGet = true)
+  private final FixedRateSwapLegConvention fixedLeg;
+  /**
+   * The market convention of the floating leg.
+   */
+  @PropertyDefinition(validate = "notNull", overrideGet = true)
+  private final InflationRateSwapLegConvention floatingLeg;
+  /**
+   * The offset of the spot value date from the trade date.
+   * <p>
+   * The offset is applied to the trade date to find the start date.
+   * A typical value is "plus 2 business days".
+   */
+  @PropertyDefinition(validate = "notNull", overrideGet = true)
+  private final BusinessDayAdjustment spotDateOffset;
+  /**
+   * An adjustment that alters the payment date by adding a period of days.
+   */
+  @PropertyDefinition(validate = "notNull", overrideGet = true)
+  private final DaysAdjustment paymentDateOffset;
+
+  //-------------------------------------------------------------------------
+
+  /**
+   * Obtains a convention based on the specified name and leg conventions.
+   * <p>
+   * The two leg conventions must be in the same currency.
+   * 
+   * @param name  the unique name of the convention 
+   * @param fixedLeg  the market convention for the fixed leg
+   * @param floatingLeg  the market convention for the floating leg
+   * @param spotDateOffset  the offset of the spot value date from the trade date
+   * @param paymentDateOffset an adjustment that alters the payment date by adding a period of days
+   * @return the convention
+   */
+  public static ImmutableFixedInflationSwapConvention of(
+      String name,
+      FixedRateSwapLegConvention fixedLeg,
+      InflationRateSwapLegConvention floatingLeg,
+      BusinessDayAdjustment spotDateOffset,
+      DaysAdjustment paymentDateOffset) {
+
+    return new ImmutableFixedInflationSwapConvention(name, fixedLeg, floatingLeg, spotDateOffset, paymentDateOffset);
+  }
+
+  //-------------------------------------------------------------------------
+  @ImmutableValidator
+  private void validate() {
+    ArgChecker.isTrue(fixedLeg.getCurrency().equals(floatingLeg.getCurrency()), "Conventions must have same currency");
```

Comment:
This error message isn't particularly helpful. Maybe this instead:

    ArgChecker.isTrue(
        fixedLeg.getCurrency().equals(floatingLeg.getCurrency()),
        Messages.format(
            "Swap leg conventions must have same currency but found {} and {}, conventions {} and {}",
            fixedLeg.getCurrency(),
            floatingLeg.getCurrency(),
            fixedLeg,
            floatingLeg));